src/components/Footer/Footer.js

```diff
@@ -53,23 +76,99 @@ const Footer = ({
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: item?.actor?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [item?.actor?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(item.id, 'post', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(item.id, 'post', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    },
+    {
+      id: 3,
+      name: 'Share link',
+      icon: <ShareBlack />,
+      onPress: onPressShare
+    }
+  ];
+
   return (
     <View style={[styles.rowSpaceBeetwen, styles.container]}>
       <View style={styles.leftGroupContainer}>
-        <TouchableOpacity testID="shareBtn" style={styles.btn} onPress={onPressShare}>
-          <View style={styles.btnShare}>
-            <MemoIc_share height={20} width={21} />
-          </View>
-        </TouchableOpacity>
+        {isShowDM && !isSelf ? (
+          <TouchableOpacity
+            testID="sendDM"
+            style={styles.btn}
+            onPress={() => refSheet.current.open()}>
+            <View style={styles.btnDM}>
+              <View style={styles.card}>
+                <MemoIc_senddm height={20} width={21} />
+                <Text style={styles.textDM}>DM</Text>
+              </View>
+            </View>
+          </TouchableOpacity>
+        ) : (
+          <TouchableOpacity testID="shareBtn" style={styles.btn} onPress={onPressShare}>
+            <View style={styles.btnShare}>
+              <MemoIc_share height={20} width={21} />
+            </View>
+          </TouchableOpacity>
+        )}
         {disableComment ? (
           <View testID="disableComment" style={styles.btn}>
-            <View style={styles.btnComment}>
+            <View style={styles.btnComment(isShowDM, isSelf)}>
               <MemoIc_comment height={24} width={25} />
             </View>
           </View>
         ) : (
           <TouchableOpacity testID="availableComment" style={styles.btn} onPress={onPressComment}>
-            <View style={styles.btnComment}>
+            <View style={styles.btnComment(isShowDM, isSelf)}>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This part of the code handles the rendering of different buttons based on certain conditions. The logic seems fine, but it's important to test thoroughly to ensure that the correct buttons are displayed under the right conditions.



<!-- This is an auto-generated comment by OSS CodeRabbit -->